src/SDKs/Azure.Core/data-plane/Azure.Core/Response.cs

```diff
@@ -12,12 +12,16 @@ public abstract class Response: IDisposable
     {
         public abstract int Status { get; }
 
-        public abstract bool TryGetHeader(string name, out string value);
-
         public abstract Stream ContentStream { get; set; }
 
         public abstract string RequestId { get; set; }
 
+        public abstract bool TryGetHeader(string name, out string value);
+
+        public abstract bool TryGetHeaderValues(string name, out IEnumerable<string> values);
+
+        public abstract bool ContainsHeader(string name);
+
```

Comment:
So many header related APIs. Maybe we should put them all on HeaderCollection struct and delegate to internal APIs on request.

Reply:
We should, but why not put it on the interface instead? `IHeaderDictionary Headers` and we can implement IHeaderDictionary on HttpClientRequest/Response internally so they don't require an additional allocation.

Reply:
Because when we have Utf8String we won't be able to add members to the interface.

Reply:
Utf8String would be .net core only and .net core would have default interface implementations.

Reply:
not necessarily. IAsyncEnumerable is not core only. And default interface members will never be portable (as it's a runtime feature), so we cannot use it.